syz-verifier/main.go

```diff
@@ -242,20 +252,84 @@ func startRPCServer(vrf *Verifier) (*RPCServer, error) {
 }
 
 // Connect notifies the RPCServer that a new Runner was started.
-func (srv *RPCServer) Connect(a *rpctype.RunnerConnectArgs, r *int) error {
+func (srv *RPCServer) Connect(a *rpctype.RunnerConnectArgs, r *rpctype.RunnerConnectRes) error {
 	srv.mu.Lock()
 	defer srv.mu.Unlock()
 	pool, vm := a.Pool, a.VM
 	srv.pools[pool].vmRunners[vm] = nil
+	if !srv.pools[pool].checked {
+		// This is the first VM connecting for this kernel so we need to find
+		// the list of unsupported system calls.
+		r.CheckUnsupportedCalls = true
+	}
+	return nil
+}
+
+// CheckUnsupported communicates to the server the list of system calls not
+// supported by the kernel corresponding to this pool and updates the list of
+// enabled system calls. This function is called once for each kernel.
+// When all kernels have reported the list of unsupported system calls, the
+// choice table will be created using only the system calls supported by all
+// kernels.
+func (srv *RPCServer) CheckUnsupported(a *rpctype.CheckUnsupportedArgs, r *int) error {
+	srv.mu.Lock()
+	defer srv.mu.Unlock()
+	if srv.pools[a.Pool].checked {
+		return nil
+	}
+	srv.pools[a.Pool].checked = true
+	vrf := srv.vrf
+
+	for _, unsupported := range a.UnsupportedCalls {
+		for c := range vrf.calls {
```

Comment:
It should be possible to do this without quadratic complexity and nested loops. Something like `if vrf.calls[target.Syscalls[unsupported.ID]]`.